discord/utils.py

```diff
@@ -108,6 +109,7 @@ def __repr__(self) -> str:
 
 
 MISSING: Any = _MissingSentinel()
+MISSING: Any = field(default_factory=_MissingSentinel())
```

Comment:
Can you provide detail about this?

Reply:
In the original code `MISSING = _MissingSentinel()` would return an error like: 
`ValueError: mutable default <class 'discord.utils._MissingSentinel'> for field is not allowed: use default_factory`
so I refer to the practice on the Internet and change 
`MISSING = _MissingSentinel()` to `MISSING: Any = field(default_factory=_MissingSentinel())` and it works

Reply:
For details, you can refer to [PEP558](https://peps.python.org/pep-0557/#mutable-default-values) shmee mentioned

Reply:
This doesn't seem to raise an error in python 3.11.0rc2, and instead appears to have the same effect as prior versions.
```pycon
Python 3.11.0rc2 (main, Sep 26 2022, 15:56:23) [Clang 14.0.0 (clang-1400.0.29.102)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from discord.utils import Any, _MissingSentinel
>>> MISSING: Any = _MissingSentinel()
>>> MISSING
...
```
```pycon
Python 3.10.6 (main, Sep 29 2022, 20:05:35) [Clang 14.0.0 (clang-1400.0.29.102)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from discord.utils import Any, _MissingSentinel
>>> MISSING: Any = _MissingSentinel()
>>> MISSING
...
```

Reply:
I applied a better fix for this in c9effafec636fc6cdd2ee50482acb464def70ec7.